src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -500,9 +500,10 @@ pub extern "C" fn scheduler_execute(
   with_scheduler(scheduler_ptr, |scheduler| {
     with_execution_request(execution_request_ptr, |execution_request| {
       with_session(session_ptr, |session| {
-        Box::into_raw(RawNodes::create(
-          scheduler.execute(execution_request, session),
-        ))
+        match scheduler.execute(execution_request, session) {
+          Ok(raw_results) => Box::into_raw(RawNodes::create(raw_results)),
+          Err(ExecutionTermination::KeyboardInterrupt) => std::ptr::null(),
```

Comment:
This feels ok as a temporary measure to avoid complicating the rust/python boundary, but it would be good to document with a NB/TODO that describes what we might want to do next here.

In particular: it seems possible that in the future we'll want to switch from returning `RawNodes` to returning a `PyResult` containing a list of results? I think that the "RawNodes" API is older than `PyResult`.

Reply:
Agreed, this is less than ideal. I'll add a comment marking this as something to change when we have a better way of communicating Errors across the FFI.

Reply:
To be clear: we have a way to communicate errors across FFI: `PyResult`. It's just that the `RawNodes` API is older than that.